pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/PinotSegmentUploadRestletResource.java

```diff
@@ -279,6 +279,10 @@ private SuccessResponse uploadSegmentInternal(FormDataMultiPart multiPart, Strin
     File tempTarredSegmentFile = null;
     File tempSegmentDir = null;
 
+    if (headers != null) {
+      LOGGER.info("Got HTTP Request headers {}", headers.getRequestHeaders());
```

Comment:
Could headers have confidential information?

Reply:
Not at this point. If there is confidential information, I'm not sure how to distinguish especially since we have internal-specific headers, and we want to keep this method generic.

Reply:
Just log the info needed (e.g. headers.getHeaderString(FileUploadDownloadClient.CustomHeaders.UPLOAD_TYPE)), but not the whole headers

Reply:
I also have segment name and table name headers specific to pbnj, I'm not looking for upload type necessarily

Reply:
The point is, the validation information will also be in headers, and cannot be logged

Reply:
+1, just log the headers that we need to log.